File: planning-idsdp-rms-rpm-backend/src/main/java/com/ibcs/idsdp/util/Response.java
package com.ibcs.idsdp.util;

import java.util.List;
import java.util.Map;

import org.springframework.data.domain.Page;

import lombok.Data;

@Data
public class Response<R> {

	private boolean success = true;
	private boolean info = false;
	private boolean warning = false;
	private String message;
	private boolean valid = false;

	//private Long id;
	private Map<String, R> model;
	private List<R> items;
	private R obj;
	private Page<R> page;
}

File: planning-idsdp-rms-frontend/src/app/main/core/models/response.ts
export interface IResponseHeader {
  requestReceivedTime: Date;
  responseTime: Date;
  hopCount: number;
  responseProcessingTimeInMs: number;
  requestId: string;
  responseCode: string;
  responseMessage: string;
  responseVersion: string;
  requestSourceService: string;
  traceId: string;
}

// tslint:disable-next-line:no-empty-interface
export interface IResponseData {
}

export interface IResponseBody<D extends IResponseData> {
    content: D[];
    pageable: {};
    totalElements: number;
}

export interface IResponse<D extends IResponseData> {
    success: boolean;
    info: boolean;
    warning: boolean;
    message: string;
    valid: boolean;
    model: any;
    items: D[];
    obj: D;
    page: any;
    totalElements: number;
}

export interface CoreResponse<D> {

    success: boolean;
    info: boolean;
    warning: boolean;
    message: string;
    valid: boolean;
    model: any;
    items: D[];
    obj: D;
    page: any;
    totalElements: number;
}

